p2p/peer.py

```diff
@@ -99,6 +103,16 @@
     from trinity.db.header import BaseAsyncHeaderDB  # noqa: F401
 
 
+# XXX: Where should this live?
```

Comment:
Yeah, that's what I wanted to avoid. Do you think it's ok to pass only the vm_configuration into `PeerPool` and `handshake()`? Or can you think of any reason why we should pass a `Chain` class?